rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/ExpressionEvaluation.kt

```diff
@@ -676,8 +676,20 @@ class ExpressionEvaluation(
     override fun eval(expression: AssignmentExpr) =
         throw RuntimeException("AssignmentExpr should be handled by the interpreter: $expression")
 
-    override fun eval(expression: GlobalIndicatorExpr) =
-        throw RuntimeException("PredefinedGlobalIndicatorExpr should be handled by the interpreter: $expression")
+    override fun eval(expression: GlobalIndicatorExpr): Value {
```

Comment:
You are iterating 3 times in order to convert a map of 99 items. into a string of 0 and 1. 
I would have implemented this algorithm in this way:
```
    val value = StringBuilder()
        for (i in IndicatorType.Predefined.range) {
            value.append(if (interpreterStatus.indicators[i] == null) "0" else if (interpreterStatus.indicators[i]!!.value) "1" else "0")
        }
        return StringValue(value.toString())
```
I tried it and it is work but maybe I makes it too easy.
What do you think about it?

Reply:
I think you are right. It's an improvement of the performance and optimization of the code.